src/cratedb/mod.rs

```diff
@@ -0,0 +1,133 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use testcontainers::{core::WaitFor, CopyToContainer, Image};
+
+const NAME: &str = "crate";
+const TAG: &str = "5.8";
+
+/// Represents a CrateDB docker instance.
+/// It is based on the official [`crate docker image`]
+///
+/// Defaults are not modified:
+/// super_user: crate
+/// password: <empty>
+/// pg_port: 5432
+/// http_port: 4200
+///
+/// You can connect run instructions by sending an HTTP call to /_sql in port 4200.
+/// [`crate http interface`]
+///
+/// or using the postgres wire compatible protocol at port 5432.
+///
+/// # Example of postgres wire.
+/// use testcontainers_modules::{postgres, testcontainers::runners::SyncRunner};
+///
+/// let postgres_instance = cratedb::CrateDB::default().start().unwrap();
+///
+/// let connection_string = format!(
+///     "postgres://crate@{}:{}/postgres",
+///     postgres_instance.get_host().unwrap(),
+///     postgres_instance.get_host_port_ipv4(5432).unwrap()
+/// );
+///
+/// let mut conn = postgres::Client::connect(connection_string, postgres::NoTls).unwrap();
+/// let rows = conn.query("SELECT 1 + 1", &[]).unwrap();
```

Comment:
currently, this is not a doctest

```suggestion
/// ```
/// use testcontainers_modules::{postgres, testcontainers::runners::SyncRunner};
///
/// let postgres_instance = cratedb::CrateDB::default().start().unwrap();
///
/// let connection_string = format!(
///     "postgres://crate@{}:{}/postgres",
///     postgres_instance.get_host().unwrap(),
///     postgres_instance.get_host_port_ipv4(5432).unwrap()
/// );
///
/// let mut conn = postgres::Client::connect(connection_string, postgres::NoTls).unwrap();
/// let rows = conn.query("SELECT 1 + 1", &[]).unwrap();
/// # assert_eq!(rows.len(), 1);
/// ```
```